config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml

```diff
@@ -56,8 +56,7 @@ presubmits:
     path_alias: "sigs.k8s.io/cluster-api-provider-azure"
     optional: false
     decorate: true
-    # please see: https://play.golang.org/p/JJSVylVPd53 for more insights
-    run_if_changed: (^[^d].*$)|(^d$)|(^.[^o].*$)|(^do$)|(^..[^c].*$)|(^doc$)|(^...[^s].*$)|(^docs$)|(^....[^/].*$)|(^[^d][^o][^c][^s]/.*$)
+    run_if_changed: '^((api|cmd|azure|config|controllers|exp|feature|hack|internal|pkg|scripts|templates|test|util|version)/|Dockerfile|Makefile|go\.mod|go\.sum|main\.go)'
```

Comment:
Do we want to include 
- `.github`
- `metadata.yaml`

And what is `cmd` dir ? Sorry, I dont see it in my local.

Reply:
The only gaps I can think of are committing Markdown files in sub-folders, like `exp/README.md`.

Fixing this in the regex might require a negative lookbehind assertion which Go doesn't support, and which takes us into inscrutable regex terroritory. Maybe we can set a [`skip_if_only_changed`](https://docs.prow.k8s.io/docs/jobs/#triggering-jobs-based-on-changes) value?

```yaml
skip_if_only_changed: "\\.md$"
```

The docs make it sound like that setting can coexist with `run_if_changed`, although I can't tell which gets precedence.

Reply:
> And what is cmd dir ? 

good catch, that was a copy paste error from the CAPI regex

> Do we want to include
.github
metadata.yaml

I don't think so? E2E isn't performing any useful validations on those files (it doesn't test github actions or PR/issue templates and it uses [a fake metadata.yaml](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/test/e2e/data/shared/v1beta1_provider/metadata.yaml))

> What if we took the skip_if_only_changed approach instead? Could be simpler here.

that's a good idea, didn't know that was a thing! Will try that

Reply:
If that's not workable, I'm fine with having a commit of `hack/observability/opentelemetry/chart/README.md` mistakenly trigger e2e tests, at least for now. Those sub-READMEs don't change often, and your current solution would be a big improvement.

Reply:
Pushed a commit with `skip_if_only_changed`. Let me know if that looks good and if so I'll squash.

/hold

Reply:
Do you think it would be worth keeping both `run_if_changed` and `skip_if_only_changed` for better readability and also since both are mutually exclusive? 
[_"run_if_changed and skip_if_only_changed are mutually exclusive."_](https://docs.prow.k8s.io/docs/jobs/#triggering-jobs-based-on-changes)

Reply:
I took "mutually exclusive" to mean "you'll get an error if you set them both."

Reply:
> Mutually exclusive is a term that describes two or more events or things that cannot happen or exist at the same time

My interpretation of mutually exclusive is that you can't have both set at the same time

Reply:
jinx @mboersma

Reply:
Sorry my bad, mutually exclusive events are not independent events.